SharedSources/MediaLibraryService.swift

```diff
@@ -284,10 +284,25 @@ extension MediaLibraryService {
 
     func fetchMedia(with mrl: URL?) -> VLCMLMedia? {
         guard let mrl = mrl  else {
+            assertionFailure("no mrl")
             return nil
         }
         return medialib.media(withMrl: mrl)
     }
+
+    func savePlaybackState(from player: VLCPlaybackController) {
+
+        let media = player.currentlyPlayingMedia
+        let mlMedia = fetchMedia(with: media.url.absoluteURL)!
```

Comment:
for some reason this call constantly frequently crashes in Medialibrary
Details 
```
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 4.1
  * frame #0: 0x0000000105a9b0df VLC for iOS`::__cxa_throw(void *, std::type_info *, void (*)(void *)) [inlined] _dispatch_once(block=<unavailable>) block_pointer) at once.h:83:39 [opt]
    frame #1: 0x0000000105a9b0df VLC for iOS`::__cxa_throw(exception_object=0x0000600001911e80, tinfo=0x000000010b075550, dest=(libc++.1.dylib`std::__1::system_error::~system_error()))(void *)) at BITCrashCXXExceptionHandler.mm:85 [opt]
    frame #2: 0x00000001078cb432 VLC for iOS`medialibrary::utils::fs::toAbsolute(path="/Users/caro/Library/Developer/CoreSimulator/Devices/34192C85-6569-4443-951B-045D45C24E39/data/Containers/Data/Application/015AE273-A6C5-4752-9C67-F072076E268E/Documents/FMA%20-%20MultiChapter.mkv") at Directory.cpp:88:9
    frame #3: 0x000000010782f90a VLC for iOS`medialibrary::factory::FileSystemFactory::createDeviceFromMrl(this=0x0000600000c0d1b8, mrl="file:///Users/caro/Library/Developer/CoreSimulator/Devices/34192C85-6569-4443-951B-045D45C24E39/data/Containers/Data/Application/015AE273-A6C5-4752-9C67-F072076E268E/Documents/FMA%2520-%2520MultiChapter.mkv") at FileSystemFactory.cpp:81:30
    frame #4: 0x0000000107656feb VLC for iOS`medialibrary::File::fromMrl(ml=0x00007fd349e00d60, mrl="file:///Users/caro/Library/Developer/CoreSimulator/Devices/34192C85-6569-4443-951B-045D45C24E39/data/Containers/Data/Application/015AE273-A6C5-4752-9C67-F072076E268E/Documents/FMA%2520-%2520MultiChapter.mkv") at File.cpp:287:30
    frame #5: 0x00000001076f17a0 VLC for iOS`medialibrary::MediaLibrary::media(this=0x00007fd349e00d60, mrl="file:///Users/caro/Library/Developer/CoreSimulator/Devices/34192C85-6569-4443-951B-045D45C24E39/data/Containers/Data/Application/015AE273-A6C5-4752-9C67-F072076E268E/Documents/FMA%2520-%2520MultiChapter.mkv") const at MediaLibrary.cpp:376:12
    frame #6: 0x000000010758f88b VLC for iOS`::-[VLCMediaLibrary mediaWithMrl:](self=0x0000600000c18000, _cmd="mediaWithMrl:", mrl="file:///Users/caro/Library/Developer/CoreSimulator/Devices/34192C85-6569-4443-951B-045D45C24E39/data/Containers/Data/Application/015AE273-A6C5-4752-9C67-F072076E268E/Documents/FMA%2520-%2520MultiChapter.mkv") at VLCMediaLibrary.m:167:54
    frame #7: 0x00000001053940b8 VLC for iOS`MediaLibraryService.fetchMedia(mrl="file:///Users/caro/Library/Developer/CoreSimulator/Devices/34192C85-6569-4443-951B-045D45C24E39/data/Containers/Data/Application/015AE273-A6C5-4752-9C67-F072076E268E/Documents/FMA%2520-%2520MultiChapter.mkv", self=0x0000600002615140) at MediaLibraryService.swift:290:25
    frame #8: 0x00000001053943fd VLC for iOS`@objc MediaLibraryService.fetchMedia(with:) at <compiler-generated>:0
    frame #9: 0x00000001052a7dbb VLC for iOS`-[VLCMovieViewController mediaForPlayingMedia:](self=0x00007fd34a04be00, _cmd="mediaForPlayingMedia:", media=0x0000600000b754f0) at VLCMovieViewController.m:1103:12
    frame #10: 0x00000001052988a1 VLC for iOS`-[VLCPlaybackController _recoverLastPlaybackStateOfItem](self=0x0000600001d06d00, _cmd="_recoverLastPlaybackStateOfItem") at VLCPlaybackController.m:1087:25
    frame #11: 0x0000000105293740 VLC for iOS`-[VLCPlaybackController observeValueForKeyPath:ofObject:change:context:](self=0x0000600001d06d00, _cmd="observeValueForKeyPath:ofObject:change:context:", keyPath=@"remainingTime", object=0x0000600001a190e0, change=0x0000600003250f80, context=0x0000000000000000) at VLCPlaybackController.m:340:13
    frame #12: 0x000000010d8683b8 Foundation`NSKeyValueNotifyObserver + 332
    frame #13: 0x000000010d86b9dc Foundation`NSKeyValueDidChange + 489
    frame #14: 0x000000010d8679c9 Foundation`NSKeyValueDidChangeWithPerThreadPendingNotifications + 143
    frame #15: 0x0000000105ac9b35 VLC for iOS`-[VLCMediaPlayer(self=0x0000600001a190e0, _cmd=<unavailable>, newMedia=<unavailable>) mediaPlayerMediaChanged:] at VLCMediaPlayer.m:1526:9 [opt]
    frame #16: 0x0000000105abcd92 VLC for iOS`-[VLCEventManager callObjectMethodWithArgs:](self=<unavailable>, _cmd=<unavailable>, message=<unavailable>) at VLCEventManager.m:350:13 [opt]
    frame #17: 0x000000011163edb5 libdispatch.dylib`_dispatch_client_callout + 8
    frame #18: 0x000000011164ce72 libdispatch.dylib`_dispatch_async_and_wait_invoke + 175
    frame #19: 0x000000011163edb5 libdispatch.dylib`_dispatch_client_callout + 8
    frame #20: 0x000000011164c080 libdispatch.dylib`_dispatch_main_queue_callback_4CF + 1540
    frame #21: 0x000000010fe3e8a9 CoreFoundation`__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
    frame #22: 0x000000010fe38f56 CoreFoundation`__CFRunLoopRun + 2310
    frame #23: 0x000000010fe38302 CoreFoundation`CFRunLoopRunSpecific + 626
    frame #24: 0x0000000112cd32fe GraphicsServices`GSEventRunModal + 65
    frame #25: 0x000000011c5dcba2 UIKitCore`UIApplicationMain + 140
    frame #26: 0x00000001052e6dc0 VLC for iOS`main(argc=1, argv=0x00007ffeea98df98) at main.m:20:16
    frame #27: 0x00000001116b3541 libdyld.dylib`start + 1
    frame #28: 0x00000001116b3541 libdyld.dylib`start + 1
```

Reply:
`FMA%20-%20MultiChapter.mkv`
Is the %20 actually part of the file name? Or is that a left over from a wrong/missing mrl decode?

Reply:
Although we changed the mrl encode/decode recently, so you might have an unmigrated database, causing this crash in the end. Could you try flushing your database?

Reply:
the %20 is not part of the file name it's a space

Reply:
This smells a lot like a double encoding happened somewhere. It should be `%20` all the way, but it starts with `%2520`, which is a `%` encoded, followed by a 20. So the percent gets decoded, but the 20 is now just a plain 20.
Where is the mrl you're passing to the medialibrary coming from?

Reply:
it directly pipes the url from a VLCMedia in there

Reply:
Please bear with my lack of iOS codebase knowledge, but ultimately, where does this MRL comes from?
TL;DR: `%2520` is very (very very) likely an encoding screw up. The question is to know where it happened.
In any case, the medialibrary is "expected" to throw an exception here, since it tries to locate a file that doesn't exist. I need to add the appropriate catch though (https://code.videolan.org/videolan/medialibrary/issues/80)

Reply:
:) sorry I will double check the internals of VLCKit to see if I find the double encoding!

Reply:
turns out this unveiled a longstanding bug in vlckit. Will be merged and fixed with the next VLCKit release